packages/marshal/src/typeGuards.js

```diff
@@ -18,32 +18,33 @@ harden(isCopyArray);
  * Check whether the argument is a pass-by-copy record, AKA a
  * "copyRecord" in @agoric/marshal terms
  *
- * @param {CopyRecord<any>} record
- * @returns {boolean}
+ * @param {Passable} record
+ * @returns {record is CopyRecord<any>}
  */
 const isRecord = record => passStyleOf(record) === 'copyRecord';
 harden(isRecord);
 
 /**
  * Check whether the argument is a remotable.
  *
- * @param {Remotable} remotable
- * @returns {boolean}
+ * @param {Passable} remotable
+ * @returns {remotable is Remotable}
  */
 const isRemotable = remotable => passStyleOf(remotable) === 'remotable';
 harden(isRemotable);
 
 /**
- * Assert that the argument is a pass-by-copy array, AKA a "copyArray"
- * in @agoric/marshal terms
- *
- * @param {CopyArray<any>} array
+ * @callback AssertArray
+ * @param {Passable} array
  * @param {string=} optNameOfArray
- * @returns {void}
+ * @returns {asserts array is CopyArray<any>}
  */
+
+/** @type {AssertArray} */
 const assertCopyArray = (array, optNameOfArray = 'Alleged array') => {
   const passStyle = passStyleOf(array);
-  assert(
+  // TODO Why does typescript want me to say "return" here?
```

Comment:
I think `return` is needed because you have a JSDoc `@returns`.

Reply:
Makes enough sense. Question removed. Thanks